File: Frontend/src/app/Model/Equipe.ts
import { Utilisateur } from "./Utilisateur";

export class Equipe {
    id! :  number ;
    nom !:string ;
    description! : string ;
    dateSupression! :string ;
    dateCreation! : string ;
    members: Utilisateur[]=[] ;

  
    static fromJson(json: any): Equipe {
    const equipe = new Equipe();
    equipe.id = json.id;
    equipe.nom =json.nom;
    equipe.description=json.description;
    equipe.dateCreation =json.dateCreation;
    equipe.dateSupression=json.dateSuppression;
    return equipe ;
    
  }





   




}

File: Backend/src/main/java/com/chahd/collabproject/entity/Equipe.java
package com.chahd.collabproject.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.ColumnDefault;

import java.time.OffsetDateTime;
import java.util.List;
import java.util.Set;

@Entity
@Data
@Getter
@Setter
@Table(name="equipe")
@NoArgsConstructor
@ToString
public class Equipe {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private int id;
    @Column(nullable = false)
    private String nom;
    private String description;
    private OffsetDateTime dateSuppression ;
    @Column(nullable = false, updatable = false)
    @ColumnDefault("now()")
    private OffsetDateTime dateCreation;
    @PrePersist
    protected void onCreate() {
        this.dateCreation = OffsetDateTime.now();
    }

    @OneToMany(mappedBy = "equipe", fetch = FetchType.LAZY)
    @JsonIgnore
    private List<MembrePole> membresPoles;

    @Transient
    @JsonIgnore
    public List<Utilisateur> getMembres() {
        return membresPoles.stream()
                .filter(mp -> mp.getDateSuppression() == null)
                .map(MembrePole::getUtilisateur)
                .toList();
    }


}
